### PR TITLE
Add option to force deploy on file validation error

### LINKED ISSUE #1245: Option to skip validation on deployment
#### Issue details

When deploying a new app, the following message sometimes (correctly) appears:
`Deploying file has failed`
`error validating "daemonset.yaml.txt": error validating data: found invalid field request for v1.ResourceRequirements; if you choose to ignore these errors, turn validation off with --validate=false`
This is impossible to do via the UI. 
Would it be useful to have a checkbox 'Ignore Validation' which optionally adds `--validate=false` to the create request so that the deployment can still be made via the UI?

##### Steps to reproduce
For example, attempt to upload the YAML described [here](https://github.com/eBay/Kubernetes/blob/master/docs/design/daemon.md)

### PR DESCRIPTION
**Pull request by floreks**
Resolves #1245 

Whenever validation error is found it will display confirmation dialog asking if user wants to deploy anyway.

zrzut ekranu z 2016-09-19 15-18-19

If solution is ok then I'll add tests.

@bryk could you take a look?

### CODE REVIEW

File: src/app/frontend/deploy/deployfromfile_controller.js [hunk 1]
```diff
@@ -29,7 +31,7 @@ export default class DeployFromFileController {
    * @param {!./../common/history/history_service.HistoryService} kdHistoryService
```
Comment (bryk) [inline src/app/frontend/deploy/deployfromfile_controller.js]: Update annotations?

File: src/app/frontend/deploy/deployfromfile_controller.js [hunk 2]
```diff
@@ -64,6 +66,8 @@ export default class DeployFromFileController {
     /** @private {!./../common/history/history_service.HistoryService} */
     this.kdHistoryService_ = kdHistoryService;
```
Comment (bryk) [inline src/app/frontend/deploy/deployfromfile_controller.js]: Add `@type`?

File: src/app/frontend/deploy/deployfromfile_controller.js [hunk 3]
```diff
@@ -98,14 +103,41 @@ export default class DeployFromFileController {
           },
           (err) => {
             defer.reject(err);  // Progress ends
-            this.log_.error('Error deploying application:', err);
-            this.errorDialog_.open('Deploying file has failed', err.data);
+            if (this.hasValidationError_(err.data)) {
+              this.handleDeployAnywayDialog_(err.data);
+            } else {
+              this.log_.error('Error deploying application:', err);
+              this.errorDialog_.open('Deploying file has failed', err.data);
```
Comment (bryk) [inline src/app/frontend/deploy/deployfromfile_controller.js]: I18n this, since you already touch the line?

**Comment by floreks** [comment]
All done. PTAL

**Comment by bryk** [comment]
Can you rebase and merge?

**Comment by floreks** [comment]
Rebased. I'll merge once travis is green.

**Pull request merged**